risc0/core/src/field/baby_bear.rs

```diff
@@ -462,28 +504,39 @@ impl field::Elem for ExtElem {
         ])
     }
 
+    /// Convert from a u64 to a base field elem, then cast to the extension field.
     fn from_u64(val: u64) -> Self {
         Self([Elem::from_u64(val), Elem::ZERO, Elem::ZERO, Elem::ZERO])
     }
 
+    // TODO(victor): Look into usages of this method.
     fn to_u32_words(&self) -> Vec<u32> {
         self.elems()
             .iter()
             .flat_map(|elem| elem.to_u32_words())
             .collect()
     }
 
+    // TODO(victor): Look into usages of this method.
     fn from_u32_words(val: &[u32]) -> Self {
         field::ExtElem::from_subelems(val.iter().map(|word| Elem(*word)))
     }
 
     // So we're not checking every subfield element every time we do
     // anything, assume that if our first subelement is valid, the
-    // whole thing is valid.  Any subfield elements will doublee check
+    // whole thing is valid.  Any subfield elements will be double checked
     // when we do operations on them anyways.
     fn is_valid(&self) -> bool {
         self.0[0].is_valid()
     }
+
+    fn is_reduced(&self) -> bool {
+        let mut valid = true;
+        for comp in self.0 {
+            valid &= comp.0 < P;
+        }
+        valid
```

Comment:
My thought on this is actually the contrary, that since during normal operation this will always evaluate to `true`, I wanted to avoid a few branches. It's definitely a micro optimization of unknown value though 🤷 Do you agree with this factoring given that, or should I go ahead and change it?